lib/gis_robot_suite.rb

```diff
@@ -108,14 +108,16 @@ def self.locate_druid_path(druid, opts = {})
   end
 
   def self.locate_esri_metadata(dir, _opts = {})
-    filename = Dir.glob("#{dir}/*.shp.xml").first # Shapefile
-    if filename.nil? || File.empty?(filename)
-      filename = Dir.glob("#{dir}/*.tif.xml").first # GeoTIFF
-      if filename.nil? || File.empty?(filename)
-        filename = Dir.glob("#{dir}/*/metadata.xml").first # ArcGRID
-        raise "Missing ESRI metadata files in #{dir}" if filename.nil? || File.empty?(filename)
-      end
+    extensions = ['shp.xml', 'tif.xml', 'metadata.xml', 'geojson.xml']
```

Comment:
This is a relatively straightforward refactor to avoid adding another if into the nest. Since they are all doing the same thing, a loop with break seemed better.